### PR TITLE
tests: add retry in metric check, make it more robust

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
in all mode test, DM-worker is just restarted before checking metric, the metric may have not been updated in time.

### What is changed and how it works?
add retry in metric check

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: tests/README.md [hunk 1]
```diff
@@ -57,5 +57,5 @@ Several convenient commands are provided:
 * `check_port_alive <PORT>` - Wrapper to check a port is alive, at most 20 times.
 * `check_port <HOST> <PORT>` - Checks a host:port is alive.
 * `wait_process_exit <process_name>` - Wait for one or more processes to exit by given process name.
-* `check_metric <PORT> <METRIC_NAME> <VALUE PATTERN LIST>...` - check metric value from prometheus.
+* `check_metric <PORT> <METRIC_NAME> <RETRY_TIME> <VALUE PATTERN LIST>...` - check metric value from prometheus.
```
Comment (csuzhangxc) [inline tests/README.md]: is `RETRY_TIME` means time with second unit (e.g. 3 means 3s) or just times without unit.

Comment (amyangfei) [reply tests/README.md]: here it means n time without second unit, each time in check_metric will wait 1s if check failed. I changed to `RETRY_COUNT`
btw if we want to use time unit, such as 1s, 1min, `RETRY_INTERVAL` may be better

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**